backend/tests/unit/filter/filter.controller.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { FilterController } from '../../../src/filter/filter.controller';
+import { FilterService } from '../../../src/filter/filter.service';
+import {
+  EdgeTypeFilterModel,
+  NodeTypeFilterModel,
+} from '../../../src/shared/filter';
+import { FilterServiceBase } from '../../../src/filter/filter.service.base';
+import {
+  getEdgeTypeFilterModelResult,
+  getNodeTypeFilterModelResult,
+} from '../../fixtures/filter/FilterQueryResults';
+
+describe('FilterController', () => {
+  let app: INestApplication;
+  const baseUrl = '/filter';
+  const mockSearchService: FilterServiceBase = {
+    getNodeTypeFilterModel: (type) => {
```

Comment:
Use nestjs's FactoryProvider for declaring your mockService. See app.controller.spec.ts for that. I don't know if its better or worse, but it's good to follow the same pattern here.

Reply:
`useValue` is fine, but maybe you could move it out of the test file into a new file.

Reply:
why not do it the same like in app.service.e2e.spec.ts? Then it's very easy to understand if someone already read app.service.e2e.spec.ts.

Reply:
There is not much difference, and since we all need to learn NestJs a bit I actually think it's quite useful to see different ways to inject stuff, especially for later when we use modules more.
Since we don't inject anything in the factory method of this (hypothetical) `FactoryProvider`, the current Provider is actually preferred.